### PR TITLE
Add vision document

### PR DESCRIPTION
**Pull request by thomaseizinger**
This document is an attempt to outline the technical vision for the codebase. I hope this increases transparency and helps documenting what problems we need to work on.

### CODE REVIEW

**Review by luckysori — COMMENTED** [review]
20/20

**Review by luckysori — APPROVED** [review]

File: docs/VISION.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+See also https://github.com/itchysats/itchysats/discussions/829.
+Current issues:
+
+- Order vs offer
+- "Settlement" interval
+- Interest vs funding rate
+
+### IO-free yet contained
+
+The execution of the protocols for contract setup and rollover are [IO-free](https://sans-io.readthedocs.io/how-to-sans-io.html).
```
Comment (klochowicz) [inline docs/VISION.md:59]: I've learned something new today, thanks for the link! 🧠  - didn't know about sans-io before

**Review by klochowicz — APPROVED** [review]
really helpful to have this all in one place, thank you!

**Pull request merged**